### ISSUE TITLE
Code improvement for RoundCubeApp.class.php

### ISSUE THREAD

**Issue by mikygee** [hypery2k/owncloud#377]
OpenBSD 6.0
Nginx
Php 5.6
OC 8.0

Hello,

I use nginx and it's chrooted, so code inside the chroot (OC) can't access /etc/ssl/openssl.cnf
But openssl_pkey_new really needs "a valid" openssl.cnf to work.

I've added some code to create an empty openssl.cnf file in tmp when it's not found (webserver chrooted)

Please review the patch

```
--- RoundCubeApp.class.php.orig Fri Feb 17 15:24:35 2017
+++ RoundCubeApp.class.php      Fri Feb 17 15:24:45 2017
@@ -116,7 +116,7 @@
             OCP\Util::writeLog('roundcube', 'OC_RoundCube_App.class.php->checkLoginData(): Found login data for oc user ' . $ocUser, OCP\Util::DEBUG);
             return $mailEntries;
         } elseif ($written == 0) {
-            OCP\Util::writeLog('roundcube', 'OC_RoundCube_App.class.php->checkLoginData(): Did not found login data for oc user ' . $ocUser, OCP\Util::DEBUG);
+            OCP\Util::writeLog('roundcube', 'OC_RoundCube_App.class.php->checkLoginData(): Did not find login data for oc user ' . $ocUser, OCP\Util::DEBUG);
             return self::writeBasicData($ocUser);
         }
     }
@@ -138,8 +138,12 @@
         /* Check if the openssl.cnf file exists */
         if (! file_exists ( $opensslcnf )) {
           OCP\Util::writeLog('roundcube', 'OC_RoundCube_App.class.php->generateKeyPair(): File '.$opensslcnf.' doesn\'t exist. Setting private_key_bits=2048', OCP\Util::ERROR);
+          /* Create an empty openssl.cnf file since it's required by openssl_pkey_new */
+          $opensslcnf = 'tmp/openssl.cnf';
+          if (touch($opensslcnf)) { OCP\Util::writeLog('roundcube', 'OC_RoundCube_App.class.php->generateKeyPair(): Empty file '.$opensslcnf.' created', OCP\Util::INFO); }
+          else { OCP\Util::writeLog('roundcube', 'OC_RoundCube_App.class.php->generateKeyPair(): Can\'t create file '.$opensslcnf, OCP\Util::ERROR); }
           /* Set a default private key length (>384) */
-          $config = array( "private_key_bits" => 2048 );
+          $config = array( "config" => $opensslcnf, "private_key_bits" => 2048 );
         }
        else
        {
```

**Comment by mckaygerhard** [comment]
hey do u tri it the pull request?

**Comment by mckaygerhard** [comment]
in the code i put a "TODO" due i suspect this behavior.. ok i'll paste lather the code by u but u can made by yourselft by using the pull request, u can made only already in github too

**Comment by mikygee** [comment]
Hello,
Can you tell me the commands I should use ?

**Comment by mckaygerhard** [comment]
.. right now i either remember but dont worry i'll research at the manpage and send good info, 

that's happened to me due much usage of the "online pull interface of github" kekekeke, 

jajajaj i feel so ironic

**Comment by mckaygerhard** [comment]
hello @mikygee right now i'm very busy at my work, next week i have two days free and i'll take more attention to this repository, i hope this gain attention also from original author

**Comment by mckaygerhard** [comment]
i now have issues, i searching for some help.. due in another machine runs good:

```
Error	 OC_USER_WEBDAVAUTH	 Invalid Url: ""	 2017-03-17T19:32:06+00:00
Error	 OC_USER_WEBDAVAUTH	 Invalid Url: ""	 2017-03-17T19:32:02+00:00
Error	 PHP	 Call to undefined function openssl_get_cert_locations() at /home/owncloud/apps/roundcube/lib/RoundCubeApp.class.php#148	 2017-03-17T19:31:48+00:00
Error	OC_USER_WEBDAVAUTH	Not possible to change password for users from web frontend using WebDAV user backend	2017-03-17T19:31:48+00:00
Error	PHP	The username is already being used at /home/owncloud/lib/private/user/manager.php#258	2017-03-17T19:31:43+00:00
Error	OC_USER_WEBDAVAUTH	Invalid Url: ""	2017-03-17T19:30:38+00:00
Error	OC_USER_WEBDAVAUTH	Invalid Url: ""
```

**Comment by mckaygerhard** [comment]
hi @mikygee here i missing for some time.. very busy with other issues and also due constants changes in owncloud and nextcloud.. both project are very bad now.. and between two i prefer nextcloud.. 

i wish to adapt the plugin to nextcloud but only nc 10 due newer versions required more updates and i'll not update nothing in my server

here the steps for made pull request here, in any case i'll commit your code this night

## Contribute coding using interface of github ##
Submitting improved code fixes and enhancements is easy:

1. Log in to GitHub

2. Fork the Repository
  * https://github.com/hypery2k/owncloud
  * Click "Fork" and you'll have your very own copy of the repository plugin source code at http://github.com/{your-username}/owncloud

3. Edit files within your fork.
  This can be done directly on GitHub.com at http://github.com/{your-username}/owncloud

4. Submit a Pull Request (tell in a issue/pull explicit details about your changes)
  * Click "Pull Prequest"
  * Enter a Message details that will go with your commit to be reviewed by others
  * Click “Send Pull Request”

### Multiple Pull Requests and Edits ###
When submitting pull requests, it is extremely helpful to isolate the changes you want included from other unrelated changes you may have made to your fork of plugin repository. The easiest way to accomplish this is to use a different branch for each pull request. The github interface when submit the pull request will ask for that. There are a number of ways to create branches within your fork, but GitHub makes the process very easy:

1. Start by finding the file you want to edit in Typesetter's code repository at https://github.com/hypery2k/owncloud.
2. Once you have located the file, navigate to the code view and click "Edit". For example, if you want to change the xxxx.php file, the "Edit" button would appear on this page: https://github.com/hypery2k/owncloud/blob/master/xxxx.php
3. Now, edit the file as you like then click "Propose File Change"

**NOTE** here we use the "xxx" file as example, but this not exist in the plugin repository

**Comment by mckaygerhard** [comment]
maybe this are the cause of #380 so need some coding fixed now...

**Comment by mckaygerhard** [comment]
this code are in pause due i searching a workaround for the php 5.3 support ue the cert file loactions function are only available for php 5.6 +

**Comment by mckaygerhard** [comment]
solved @mikygee now please test and report feedback
